_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,392 @@
+---
+layout: default
+title: Connecting to other ML platforms
+has_children: false
+nav_order: 60
+---
+
+# Connecting to other ML platforms
+
+Machine Learning (ML) Connectors enables the ability to integrate OpenSearch's ML capabilities with third-party ML tools and platforms. Through connectors, OpenSearch can invoke these third-party endpoints to enrich query results and data pipelines.
+
+## Supported connectors
+
+As of OpenSearch 2.9, connectors have been tested for the following ML tools, though its possible to make connectors for tools not listed:
+
+- [Amazon SageMaker](https://aws.amazon.com/sagemaker/) allows you to host and manage the life cycle of text embedding models on Amazon SageMaker, powering semantic search queries on OpenSearch. When connected, Amazon SageMaker hosts your models and OpenSearch is used to query inferences. This benefits Amazon SageMaker users who value it's functionality such as model monitoring, serverless hosting, and workflow automation for continuous training and deployment.
+- [Amazon Comprehend](https://aws.amazon.com/comprehend/) allows you to extract metadata automatically from OpenSearch ingest pipeline, enriching data indexed in OpneSearch and improving your search capabilities. 
+- [ChatGPT](https://openai.com/blog/chatgpt) enables you to run OpenSearch queries while invoking the ChatGPT API, helping you build on OpenSearch faster and improves the speed in which OpenSearch search functionality can retrieve data.
+- [Amazon Bedrock](https://aws.amazon.com/bedrock/) is an alternative to ChatGPT inside a fully managed-service.
+- [NVIDIA Triton](https://developer.nvidia.com/triton-inference-server) allows you to host text embedding models on NVIDIA's high performance model serving technology. Power semantic search queries and vector generation ingest pipelines using [GPU-acceleration]({{site.url}}{{site.baseurl}}/ml-commons-plugin/gpu-acceleration/).
+- [Cohere](https://cohere.com/) allows you to use data from OpenSearch to power Cohere's large language models.
+
+## Prerequisites
+
+If you are an admin deploying wanting to use an ML connector, make sure that target model of the connector has already been deployed on your chosen platform. Furthermore, make sure that you have access to send and receive data to the third-party API for your connector. 
+
+When access control is enabled on your third-party platform, you can enter your security settings using the `authorization` or `credential` settings inside the connector API.
+
+### Adding trusted endpoints
+
+To make sure that you configure connectors in OpenSearch, add the trusted endpoints into your cluster settings using the `plugins.ml_commons.trusted_connector_endpoints_regex` setting, as shown in the following example:
+
+```
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+            "^https://runtime\\.sagemaker\\..*\\.amazonaws\\.com/.*$",
```

Comment:
So a forward slash does not need to be escaped with `\\`?

Reply:
@Zhangxunmt: Do we need to adjust this example in any way?

Reply:
A forward slash doesn't need to be escaped in java regex, we don't need to adjust this example.

Reply:
Thanks. I think we should specify that it's a Java regex that's used here because regex expressions can be of many flavors.